src/python/library/tritonclient/http/_utils.py

```diff
@@ -37,12 +37,14 @@ def _get_error(response):
     if response.status_code != 200:
         body = response.read()
```

Comment:
Small tweaks: for the message what if keep to the server message or "" and we can put the parsing exception and body in debug details (only if there is a parsing error). In that way the message is always coming from the server and if there is a parsing error it is passed in debug details. @Tabrizian, @tgerdesnv what do you think?

@brightsparc thanks for you work on this!

```
if response.status_code != 200:
   status_string = str(response.status_code)
   body = None
   try:
       body = response.read().decode()
       error_message = json.loads(body)["error"] if body else ""
       return InferenceServerException(msg=error_message, status=status_string)
   except e:
       return InferenceServerException(msg="",status =status_string,debug_details=f"Exception {e} while parsing {body}")
```

Reply:
We don't have a unified way to do that. I think parsing + swallowing exceptions to allow for more context makes sense.

Reply:
I've updated my PR to return empty message, and set the debug details as indicated above.

I've left the decode outside of the try/except as this shouldn't fail, and left catching the explicit json.JSONDecodeError error, so any other errors should get thrown.

Reply:
sorry for the delayed response. I think we could probably use `debug_details` for the body but I think the JSONDecode error message should be part of the `msg`.

Reply:
That is what I hard last commit, and changed based on the feedback from [nnshah1](https://github.com/nnshah1) changed it again.

Can you confirm this is what you would like to see @Tabrizian 

```
        except json.JSONDecodeError as e:
            return InferenceServerException(msg=str(e),
                                            status=str(response.status_code),
                                            debug_details=body.decode())
```

Reply:
@Tabrizian and I have discussed offline and he will follow up here - sorry for the thrash / back and forth.